### PR TITLE
Annotate all tests with #[kani::proof]

### PR DESCRIPTION
**Pull request by tedinski**
### Description of changes: 

80% of these changes were from:

```
shopt -s globstar

for file in **/*.rs; do
    # there were still some pub main?
    sed -i 's/pub fn main/fn main/g' "$file"
    # add proof harness annotation to all
    sed -i 's/fn main/#[kani::proof]\nfn main/g' "$file"
done
```

But there were additional manual changes needed under `tests/expected` and `tests/ui` to fix issues with things like line numbers.

### Call-outs:

* This patch is required to allow me to change the default behavior to "run all proof harnesses" and also to remove the hack to only sometimes provide `--crate-type=lib`. Those changes are not in this PR. I've separated this monster out.

* Alt-click on the files in the github code review ui will expand or collapse all files, allowing you you quickly look at only certain files.

### Testing:

* How is this change tested? existing suite

* Is this a refactor change? yes

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: tests/expected/assert-location/assert-false/expected [hunk 1]
```diff
@@ -1,3 +1,3 @@
-line 17 assertion failed: false: FAILURE
```
Comment (zhassan-aws) [inline tests/expected/assert-location/assert-false/expected:1]: AFAIK, this test is intentionally checking for line number since there was a related bug that @celinval fixed.

File: tests/expected/assert-location/debug-assert/expected [hunk 2]
```diff
@@ -1,2 +1,2 @@
-line 12 "This should fail and stop the execution": FAILURE
```
Comment (zhassan-aws) [inline tests/expected/assert-location/debug-assert/expected:1]: Same for this one.

File: tests/kani/Assert/multiple_asserts.rs [hunk 3]
```diff
@@ -4,6 +4,7 @@
 // kani-verify-fail
 // Check that this doesn't trigger a fake loop. See issue #636.
 #[no_mangle]
```
Comment (zhassan-aws) [inline tests/kani/Assert/multiple_asserts.rs:6]: Remove no_mangle?

File: tests/ui/arguments-proof/expected [hunk 4]
```diff
@@ -1,6 +1 @@
-  --> main.rs:13:1
```
Comment (zhassan-aws) [inline tests/ui/arguments-proof/expected:1]: Why were these removed?

File: tests/ui/cbmc_checks/float-overflow/check_message.rs [hunk 5]
```diff
@@ -11,6 +11,7 @@ fn dummy(result: f32) -> f32 {
     result.round()
 }
 
+#[kani::proof]
```
Comment (zhassan-aws) [inline tests/ui/cbmc_checks/float-overflow/check_message.rs:14]: Is the duplicate annotation intentional?

File: tests/ui/cbmc_checks/pointer/check_message.rs [hunk 6]
```diff
@@ -7,6 +7,7 @@ fn not_zero(p1: *const i32) {
     assert!(unsafe { *p1 != 0 });
 }
 
+#[kani::proof]
```
Comment (zhassan-aws) [inline tests/ui/cbmc_checks/pointer/check_message.rs:10]: Same (Is the duplicate annotation intentional?)

File: tests/ui/cbmc_checks/signed-overflow/check_message.rs [hunk 7]
```diff
@@ -13,6 +13,7 @@ fn dummy(var: i32) {
 }
 
 
+#[kani::proof]
```
Comment (zhassan-aws) [inline tests/ui/cbmc_checks/signed-overflow/check_message.rs:16]: Here as well and in the next few tests.

File: tests/ui/regular-output-format-fail/expected [hunk 8]
```diff
@@ -1,4 +1,2 @@
-Check 1: main.assertion.2
 Description: "assertion failed: 1 + 1 == 3"
-Description: "pointer to dead object"
```
Comment (zhassan-aws) [inline tests/ui/regular-output-format-fail/expected:3]: Is this check no longer produced with `--crate-type lib`?

File: tests/ui/terse-output-format-pass/expected [hunk 9]
```diff
@@ -1,2 +1 @@
 VERIFICATION RESULT:
-** 0 of 3 failed
```
Comment (zhassan-aws) [inline tests/ui/terse-output-format-pass/expected:2]: This is now pretty much not doing any checking. Should we add this line back, or at least its prefix `** 0 of`?

File: tests/ui/unsupported-features/thread/main.rs [hunk 10]
```diff
@@ -7,6 +7,7 @@ thread_local! {
     static COND : bool = kani::any();
 }
 
+#[kani::proof]
```
Comment (zhassan-aws) [inline tests/ui/unsupported-features/thread/main.rs:10]: Duplicate annotation.

**Review by zhassan-aws — COMMENTED** [review]
It's unfortunate to not be able to run Kani directly on `main` without an annotation :(

File: tests/ui/regular-output-format-fail/expected (hunk 8 above)
Comment (tedinski) [reply tests/ui/regular-output-format-fail/expected:3]: correct

File: tests/ui/terse-output-format-pass/expected (hunk 9 above)
Comment (tedinski) [reply tests/ui/terse-output-format-pass/expected:2]: My interpretation was this test was mostly "run the output format at all" and not actively really looking for any results here.

File: tests/ui/arguments-proof/expected (hunk 4 above)
Comment (tedinski) [reply tests/ui/arguments-proof/expected:1]: Line number sensitivity.

**Review by zhassan-aws — APPROVED** [review]
